test/init.spec.js

```diff
@@ -32,4 +35,101 @@ describe('linter-js-standard-engine', function () {
       return expect(data, 'to be empty')
     })
   })
+  it('should skip files with the wrong scope', () => {
+    const textEditor = textEditorFactory({ source: '# markdown', scopeName: 'source.gfm' })
+    return expect(lint(textEditor), 'to be fulfilled').then(data => {
+      expect(data, 'to be empty')
+    })
+  })
+  it('should not skip any files if the ignore option is not set', () => {
+    const filePath = path.resolve(__dirname, 'fixtures', 'simpleSemiStandard', 'index.js')
+    const textEditor = textEditorFactory({
+      source: fs.readFileSync(filePath),
+      path: filePath
+    })
+    return expect(lint(textEditor), 'to be fulfilled').then(function (data) {
+      return expect(data, 'to be a valid lint report')
+    })
+  })
+  it('should clean linters when deactivated', () => {
+    let cleaned = false
+    const plugin = proxyquire('../init', {
+      './lib/getLinter': {
+        cleanLinters () {
+          cleaned = true
+        }
+      }
+    })
+
+    expect(cleaned, 'to be false')
+    plugin.deactivate()
+    expect(cleaned, 'to be true')
+  })
+
+  describe('error handling', () => {
+    let currentError
+    const linter = proxyquire('../init', {
+      './lib/lint' () {
+        return () => {
+          return Promise.reject(currentError)
+        }
+      }
+    }).provideLinter()
+
+    for (const msg of [
+      'no supported linter found',
+      'no package.json found',
+      'Could not load linter "foo"'
+    ]) {
+      it(`should suppress "${msg}" errors`, () => {
+        currentError = new Error(msg)
+        return expect(linter.lint(textEditorFactory('')), 'to be fulfilled').then(function (data) {
+          return expect(data, 'to be empty')
+        })
+      })
+    }
+
+    it('should add errors that are not suppressed', () => {
+      atom.notifications._errors = []
+      currentError = new Error('do not suppress me')
+      return expect(linter.lint(textEditorFactory('')), 'to be fulfilled').then(function (data) {
+        expect(data, 'to be empty')
+
+        expect(atom.notifications._errors, 'to have length', 1)
+        const actual = atom.notifications._errors[0]
+
+        actual.silence()
+
+        expect(actual, 'to satisfy', {
+          desc: expect.it('to be', 'do not suppress me'),
+          obj: expect.it('to satisfy', {
+            error: expect.it('to be', currentError),
+            detail: expect.it('to be', currentError.stack),
+            dismissable: expect.it('to be true')
+          })
+        })
```

Comment:
I appreciate the effort using unexpected! I just have a few points comments here, in order to make your experience a little more pleasant :-) It's nitpicky - the tests are working fine - so it's only stylistic input.

---

You don't have to use `expect.it` when using the `to satisfy` assertion.

`to satisfy` can do all sorts of magic, but you can also use it like a forgiving version of `to equal`.

```js
expect({ foo: 'bar', qux: 'baz' }, 'to equal', { foo: 'bar' });

// Fails as the subject has an additional property qux which is missing on the expected value.

expect({ foo: 'bar', qux: 'baz' }, 'to satisfy', { foo: 'bar' });

// Passes
```

The reason the example with `to satisfy` passes is that instead of comparing the equality of the two objects, it is used as a sort of "specification". The value compared must be an object with these properties - extra properties are ignored.

The test case I comment on here could be written as:

```js
expect(actual, 'to satisfy', {
  desc: 'do not suppress me',
  obj: {
    error: currentError,
    detail: currentError.stack,
    dismissable: true
  }
})
```

If you actually mention all the properties on the atom notification object, we could change `to satisfy` to `to equal`.